assets/js/modules/analytics/components/dashboard/DashboardAllTrafficWidgetV2/index.js

```diff
@@ -69,6 +139,30 @@ function DashboardAllTrafficWidget() {
 
 	const serviceReportURL = useSelect( ( select ) => select( MODULES_ANALYTICS ).getServiceReportURL( reportType, reportArgs ) );
 
+	useEffect( () => {
+		if ( dateRange !== currentRange ) {
+			setFirstLoad( true );
+			setCurrentRange( dateRange );
+		} else if ( pieChartLoaded && totalUsersLoaded && userCountGraphLoaded ) {
+			setFirstLoad( false );
+		}
+	}, [
+		pieChartLoaded,
+		totalUsersLoaded,
+		userCountGraphLoaded,
+		dateRange,
+		currentRange,
+	] );
+
+	if ( pieChartError ) {
+		return <ReportError moduleSlug="analytics" error={ pieChartError } />;
+	}
+
+	const totalUsers = parseInt( pieChartReport?.[ 0 ]?.data?.totals?.[ 0 ]?.values?.[ 0 ] );
+	if ( pieChartLoaded && ( Number.isNaN( totalUsers ) || totalUsers < 1 ) ) {
```

Comment:
Why not use `isZeroReport( pieChartReport )`? That's the intended function for this purpose - we shouldn't manually inspect the response to check for a "zero data" state.